watson/watson.py

```diff
@@ -247,6 +248,13 @@ def start(self, project, tags=None):
                 )
             )
 
+        default_tags = self._default_tags(project)
+        if default_tags and not restart:
+            if tags is None:
+                tags = default_tags
+            else:
+                tags = tags + default_tags
```

Comment:
I find this overly complicated or I don't get it. Why do you look up the element only in the sublist `sequence[:index]`? If this some kind of optimization, I don't understand how it's supposed to work, and, anyway, I think creating many copies of the list defeats the purpose.